File: agents/android/app/src/main/java/com/example/anton2319sdocsonwg/Peer.java
package com.example.anton2319sdocsonwg;

import com.fasterxml.jackson.databind.ObjectMapper;

public class Peer {
    public String preshared_key;
    public String endpoint;
    public String[] allowed_ips;
    public String id;

    public static Peer fromByteArray(byte[] byteArray) throws Exception {
        ObjectMapper mapper = new ObjectMapper();
        return mapper.readValue(byteArray, Peer.class);
    }
}

File: tunnel/peer.go
package tunnel

import (
	"encoding/json"
)

type Peer struct {
	PublicKey    string   `json:"public_key"`
	PresharedKey string   `json:"preshared_key"`
	Endpoint     string   `json:"endpoint"`
	AllowedIPs   []string `json:"allowed_ips"`
	Id           string   `json:"id"`
}

func (p *Peer) ToByteArray() []byte {
	// Marshal the Peer struct into a byte array
	byteArray, err := json.Marshal(p)
	if err != nil {
		return nil
	}

	return byteArray
}

func FromByteArray(byteArray []byte) (*Peer, error) {
	p := &Peer{}
	err := json.Unmarshal(byteArray, p)
	if err != nil {
		return nil, err
	}

	return p, nil
}
